File: Data Structures & Algorithms/counting-bits/submission-1.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        def countBit(d: int) -> int:
            result = 0
            while d > 0:
                result += (d & 1)
                d >>= 1
            return result
        r = []
        for i in range(n + 1):
            r.append(countBit(i))
        
        return r

File: Data Structures & Algorithms/subsets/submission-1.cs
public class Solution {
    List<List<int>> result;
    public List<List<int>> Subsets(int[] nums) {
        List<int> subset = new List<int>();
        result = new List<List<int>>();
        Subsets(nums, 0, subset);
        return result;
    }

    public void Subsets(int[] nums, int index, List<int> subset)
    {
        result.Add(new List<int>(subset));

        for(int i = index; i < nums.Length; i++)
        {
            subset.Add(nums[i]);            
            Subsets(nums, i + 1, subset);
            subset.RemoveAt(subset.Count - 1);
        }
    }
}
